src/plugins/external_tool_support/src/blast/align_worker_subtasks/BlastAlignToReferenceTask.cpp

```diff
@@ -0,0 +1,371 @@
+                                                     DbiDataStorage *_storage,
+                                                     const QString &taskNameSuffix)
+    : Task(tr("Align reads with BLAST & Smith-Waterman task") + taskNameSuffix, TaskFlags_NR_FOSE_COSC),
+      dbPath(_blastDbPath),
+      reads(_reads),
+      reference(_reference),
+      storage(_storage) {
+    QScopedPointer<U2SequenceObject> referenceSequence(StorageUtils::getSequenceObject(storage, reference));
+    CHECK_EXT(referenceSequence->getSequenceLength() < INT_MAX, setError(tr("Maximum supported reference sequence length is 2Gb")), );
+    setMaxParallelSubtasks(MAX_PARALLEL_SUBTASKS_AUTO);
+}
+
+void BlastAlignToReferenceTask::prepare() {
+    CHECK(!reads.isEmpty(), );
+
+    QScopedPointer<U2SequenceObject> referenceSequence(StorageUtils::getSequenceObject(storage, reference));
+    CHECK_EXT(!referenceSequence.isNull(), setError(tr("Reference sequence is not found")), );
+
+    BlastTaskSettings settings;
+    settings.programName = "blastn";
+    settings.databaseNameAndPath = dbPath;
+    settings.wordSize = 11;
+    settings.xDropoffGA = 20;
+    settings.xDropoffUnGA = 10;
+    settings.xDropoffFGA = 100;
+    settings.numberOfProcessors = AppContext::getAppSettings()->getAppResourcePool()->getIdealThreadCount();
+    settings.numberOfHits = 1;  // 'culling_limit' - keep the longest hit.
+    settings.gapOpenCost = 2;
+    settings.gapExtendCost = 2;
+    for (const SharedDbiDataHandler &read : qAsConst(reads)) {
+        QScopedPointer<U2SequenceObject> readObject(StorageUtils::getSequenceObject(storage, read));
+        CHECK_EXT(!readObject.isNull(), setError(L10N::nullPointerError("U2SequenceObject")), );
+
+        QByteArray readSequence = readObject->getWholeSequenceData(stateInfo);
+        CHECK_OP(stateInfo, );
+        bool isEmptyReadSequence = std::all_of(readSequence.begin(), readSequence.end(), [](char c) { return c == U2Msa::GAP_CHAR || c == 'N'; });
+        CHECK_EXT(!isEmptyReadSequence, setError(tr("Read doesn't contain meaningful data")), );
+        settings.querySequences << readSequence;
+
+        const DNAAlphabet *readAlphabet = readObject->getAlphabet();
+        if (settings.alphabet == nullptr) {
+            settings.alphabet = readAlphabet;
+        } else {
+            const DNAAlphabet *commonAlphabet = U2AlphabetUtils::deriveCommonAlphabet(settings.alphabet, readAlphabet);
+            if (commonAlphabet != settings.alphabet) {
+                CHECK_EXT(commonAlphabet != nullptr, setError(tr("Failed to derive common alphabet for all reads")), );
+                settings.alphabet = commonAlphabet;
+            }
+        }
+    }
+    CHECK_EXT(settings.alphabet->isNucleic(), setError(tr("Can't run alignment on non-nucleic reads")), );
+    settings.isNucleotideSeq = true;
+    settings.needCreateAnnotations = false;
+    settings.groupName = "blast";
+    CHECK_OP(stateInfo, );
+    settings.outputType = 5;
```

Comment:
What is `5`?

Reply:
And other numbers above

Reply:
Part of the original code. 5 is a constant from the original code that specifies output format (we support options 5 and 6)
https://www.ncbi.nlm.nih.gov/books/NBK279690/pdf/Bookshelf_NBK279690.pdf

I will refactor BLAST settings and introduce enums in a separate PR